syntax = "proto3";

option java_multiple_files = true;
option java_package = "c.pxsdirac.grpc.tushare";
option java_outer_classname = "TushareService";
option objc_class_prefix = "RTG";

package routeguide;

// Interface exported by the server.
service TushareService {
    rpc MoneySupply(EmptyParam) returns (Dataframe) {}
    rpc HistData(HistDataRequest) returns (Dataframe) {}
    rpc TickData(TickDataRequest) returns (Dataframe) {}

}

message EmptyParam{

}

message StringValue{
    string value = 1;
}
message HistDataRequest{
    string code = 1;
    Date start = 2;
    Date end = 3;
    KType kType = 4;
}
message TickDataRequest{
    string code = 1;
    Date date = 2;
}
message Dataframe{
    string json = 1;
}

message Date{
    int32 year = 1;
    int32 month = 2;
    int32 day = 3;
}

enum KType{
    D = 0;
    W = 1;
    M = 2;
    m5 = 3;
    m15 = 4;
    m30 = 5;
    m60 = 6;
}
